src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,426 @@
+            .withOauthToken(Environment.PROMETHEUS_WEB_CLIENT_ACCESS_TOKEN)
+            .withTrustCerts(true)
+            .build();
+        log.info("init openshift client");
+        oc = new KubernetesClientBuilder().withConfig(config).build().adapt(OpenShiftClient.class);
+
+        log.info("init cluster capacity info");
+        maxStandardStreamingUnitsInCluster = FleetshardUtils.getClusterCapacityFromMKAgent(oc, ManagedKafkaType.standard);
+        log.debug("Max standard streaming units according to MKAgent capacity '{}'", maxStandardStreamingUnitsInCluster);
+        maxDeveloperStreamingUnitsInCluster = FleetshardUtils.getClusterCapacityFromMKAgent(oc, ManagedKafkaType.developer);
+        log.debug("Max developer streaming units according to MKAgent capacity '{}'", maxDeveloperStreamingUnitsInCluster);
+
+        log.info("init kafka management");
+        var apps = ApplicationServicesApi.applicationServicesApi(
+            Environment.PRIMARY_USERNAME,
+            Environment.PRIMARY_PASSWORD);
+        kafkaMgmtApi = apps.kafkaMgmt();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown() {
+
+        // delete kafka instance
+        try {
+            KafkaMgmtApiUtils.cleanKafkaInstance(kafkaMgmtApi, KAFKA_INSTANCE_NAME);
+        } catch (Throwable t) {
+            log.error("clean main kafka instance error: ", t);
+        }
+    }
+
+    @DataProvider(name = "managedKafkaTypes")
+    public Object[][] managedKafkaTypeDataProvider() {
+        return new ManagedKafkaType[][]{
+            {ManagedKafkaType.standard},
+            {ManagedKafkaType.developer}};
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    @SneakyThrows
+    public void testReservedDeploymentExistence(ManagedKafkaType mkType) {
+
+        String namespace = String.format("reserved-kafka-%s-1", mkType);
+        log.info("evaluating namespace '{}'", namespace);
+
+        String managedKafkaName = String.format("reserved-kafka-%s-1", mkType);
+        log.info("managed kafka reserved deployment '{}'", managedKafkaName);
+        Optional<ManagedKafka> mkOptional = FleetshardUtils.managedKafka(oc).inNamespace(namespace).list().getItems().stream().filter(ManagedKafka::isReserveDeployment).findAny();
+        Assert.assertTrue(mkOptional.isPresent(), "reserved deployment is not present");
+
+        log.info("assert existence of resources which are part of reserved deployment");
+        int expectedZookeeperPodsCount;
+        int expectedKafkaPodsCount;
+        switch (mkType.name()) {
+            case "standard":
+                expectedZookeeperPodsCount = 3;
+                expectedKafkaPodsCount = 3;
+                break;
+            case "developer":
+                expectedZookeeperPodsCount = 1;
+                expectedKafkaPodsCount = 1;
+                break;
+            default:
+                throw new Exception("unsupported managed kafka type");
+        }
```

Comment:
```suggestion
        switch (mkType) {
            case standard:
                expectedZookeeperPodsCount = 3;
                expectedKafkaPodsCount = 3;
                break;
            case developer:
                expectedZookeeperPodsCount = 1;
                expectedKafkaPodsCount = 1;
                break;
            default:
                throw new Exception("unsupported managed kafka type");
        }
```

nitpick: I think you could do this throughout the switch statement

Reply:
thanks i never noticed this 👌